openrouteservice/src/main/java/heigit/ors/isochrones/builders/concaveballs/ConcaveBallsIsochroneMapBuilder.java

```diff
@@ -524,4 +537,41 @@ private void copyConvexHullPoints(Polygon poly)
 			prevIsoPoints.add(new Coordinate(p.getX(), p.getY()));
 		}
 	}
+
+
+    /**
+     * Determines mean speed to calculate the reachfactor property.
+     *
+     * @param encoder The relevant {@link FlagEncoder} instance.
+     * @return mean speed constant
+     */
+    private double getMeanSpeed(FlagEncoder encoder){
```

Comment:
It would probably be better to move this into the ORSAbstractFlagEncoder object and then force each flagencoder to override it. Then inside the getMeanSpeed method you just need to check if it is an instance of ORSABstractFlagEncoder, and if so then call the method from there to get the speed:
```java
if (encoder instanceof ORSAbstractFlagEncoder) {
  meanSpeed = ((ORSAbstractFlagEncoder) encoder).getMeanSpeed();
}
```
That way, you don't have to add another if statement here everytime we add a new FlagEncoder.

Reply:
ah nice! yes, that's what I was looking for! Will do tmrw

Reply:
hope that's better!